lib/caption-stream.js

```diff
@@ -177,17 +177,48 @@
   // semantics see
   // http://www.gpo.gov/fdsys/pkg/CFR-2010-title47-vol1/pdf/CFR-2010-title47-vol1-sec15-119.pdf
   var PADDING                    = 0x0000,
+
+      // Pop-on Mode
       RESUME_CAPTION_LOADING     = 0x1420,
+      END_OF_CAPTION             = 0x142f,
+
+      // Paint-on Mode
+      RESUME_DIRECT_CAPTIONING   = 0x1429,
+
+      // Roll-up Mode
+      ROLL_UP_2_ROWS             = 0x1425,
+      ROLL_UP_3_ROWS             = 0x1426,
+      ROLL_UP_4_ROWS             = 0x1427,
+      CARRIAGE_RETURN            = 0x142d,
+
+      // Erasure
       ERASE_DISPLAYED_MEMORY     = 0x142c,
-      ERASE_NON_DISPLAYED_MEMORY = 0x142e,
-      END_OF_CAPTION             = 0x142f;
+      ERASE_NON_DISPLAYED_MEMORY = 0x142e;
+
+  // the index of the last row in a CEA-608 display buffer
+  var BOTTOM_ROW = 14;
+  // CEA-608 captions are rendered onto a 34x15 matrix of character
+  // cells. The "bottom" row is the last element in the outer array.
+  var createDisplayBuffer = function() {
+    var result = [], i = BOTTOM_ROW + 1;
+    while (i--) {
+      result.push('');
```

Comment:
any particular reason this is an array of strings rather than an array of arrays?